crates/iota-json-rpc-types/src/iota_transaction.rs

```diff
@@ -2356,12 +2358,41 @@ impl Filter<EffectsWithInput> for TransactionFilter {
                     && (module.is_none() || matches!(module,  Some(m2) if m2 == &m.to_string()))
                     && (function.is_none() || matches!(function, Some(f2) if f2 == &f.to_string()))
             }),
-            TransactionFilter::TransactionKind(kind) => item.input.kind().to_string() == *kind,
-            TransactionFilter::TransactionKindIn(kinds) => {
-                kinds.contains(&item.input.kind().to_string())
-            }
+            TransactionFilter::TransactionKind(kind) => IotaTransactionKind::from_str(kind)
+                .map(|kind| IotaTransactionKind::from(item.input.kind()) == kind)
+                .unwrap_or(false),
+            TransactionFilter::TransactionKindIn(kinds) => kinds.iter().any(|kind| {
+                IotaTransactionKind::from_str(kind)
+                    .map(|kind| IotaTransactionKind::from(item.input.kind()) == kind)
+                    .unwrap_or(false)
+            }),
             // this filter is not supported, RPC will reject it on subscription
             TransactionFilter::Checkpoint(_) => false,
         }
     }
 }
+
+/// Represents the type of a transaction.
+#[derive(Debug, Clone, Copy, PartialEq, Eq, EnumString, Display)]
+#[non_exhaustive]
+pub enum IotaTransactionKind {
+    Genesis = 0,
+    ConsensusCommitPrologueV1 = 1,
+    AuthenticatorStateUpdateV1 = 2,
+    RandomnessStateUpdate = 3,
+    EndOfEpochTransaction = 4,
+    ProgrammableTransaction = 5,
```

Comment:
This is a great improvement. 

I think however it is best to maintain backward compatibility with what is currently stored in the database, so that we don't require resetting and resyncing.

So I would suggest making adding the new variants following the former `iota_indexer::types::TransactionKind` `enum`, e.g.

```suggestion
    SystemTransaction = 0,
    ProgrammableTransaction = 1,
    Genesis = 2,
    ConsensusCommitPrologueV1 = 3,
    AuthenticatorStateUpdateV1 = 4,
    RandomnessStateUpdate = 5,
    EndOfEpochTransaction = 6,

```

When storing the kind we should use `kind > 1`, but for read queries we should support all including `0`. `0` should get the rows from `tx_kinds` where `kind = 0` **AND** `kind > 1`.

What do you think?

Reply:
Good point, yes we should do this! And by this we can also get all future `SystemTransactions` in a more efficient way!

Reply:
Adjusting the SQL..

Reply:
Fixed with https://github.com/iotaledger/iota/pull/5434/commits/6487fc3eeef8dce45321a2074312e7b0c87468fe

Needs however still some testing.